lib/client/presence/local-doc-presence.js

```diff
@@ -40,7 +40,13 @@ LocalDocPresence.prototype.destroy = function(callback) {
   this._doc.removeListener('load', this._loadHandler);
   this._doc.removeListener('destroy', this._destroyHandler);
 
-  LocalPresence.prototype.destroy.call(this, callback);
+  var presence = this;
+  LocalPresence.prototype.destroy.call(this, function(error) {
+    // If the doc does not exist, let's just ignore the error. There's no need to
+    // tidy up presence on a deleted doc.
+    if (error && error.code === ERROR_CODE.ERR_DOC_DOES_NOT_EXIST) error = null;
```

Comment:
As discussed, doing the skip here means that if `!this._doc.type` (the check in `LocalDocPresence.prototype.submit`), the `LocalPresence.prototype.destroy.call` could have been replaced with a short-circuit.

But you said that you wanted the cleanup here to still run:
https://github.com/share/sharedb/blob/a81e2c99af77da12885f0fa860e87cc2602a6cc1/lib/client/presence/local-presence.js#L39

So the update to instead not emit an error in `LocalDocPresence.prototype.submit` if the value is `null` seems good.